src/main/java/org/aszjch/demoapp/aplication/MapperHelper.java

```diff
@@ -0,0 +1,32 @@
+package org.aszjch.demoapp.aplication;
+
+import lombok.RequiredArgsConstructor;
+import org.aszjch.demoapp.infrastructure.storage.TempFilesService;
+import org.mapstruct.Named;
+import org.springframework.stereotype.Component;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.File;
+import java.io.IOException;
+import java.time.Clock;
+import java.time.OffsetDateTime;
+
+@RequiredArgsConstructor
+@Named("MapperHelper")
+@Component
+class MapperHelper {
+
+    private final Clock clock;
+    private final TempFilesService tempFilesService;
+
+    @Named("getCurrentTime")
+    OffsetDateTime getCurrentTime(Object ignore) {
```

Comment:
Czy ten Object ignore jest na potrzeby mapstructa?

Reply:
Yup, mapstruct jest trochę upośledzony pod tym względem i wymaga przy mapowaniu, żeby wepchnąć do metody jakieś source. Szukałem wcześniej po necie, to piszą, że się nie da obejść dziadostwa, przynajmniej jeżeli chcę użyć dependency injection opartego o Named/QualifiedByName